src/Renderer/ThreeExtended/GeoJSON2Three.js

```diff
@@ -39,66 +40,82 @@ function readCoordinates(crsIn, crsOut, coordinates) {
         }
+        // expand extent if present
+        if (extent) {
+            extent.expandByPoint(out[out.length - 1]);
+        }
     }
+
     return out;
 }
 
-// Helper struct that returns an object { type: "", vertices: [...], indices: [...]}:
+// Helper struct that returns an object { type: "", vertices: [...], indices: [...], extent}:
 // - type is the geom type
 // - vertices is an array of THREE.Vector3
 // - indices is optional, and are currently only used for polygons
+// - extent is optional, it's extent vertices
 // Multi-* geometry types are merged in one.
 const GeometryToVertices = {
-    point(crsIn, crsOut, coordinates, filteringExtent) {
-        const coords = readCoordinates(crsIn, crsOut, coordinates);
+    point(crsIn, crsOut, coordinates, filteringExtent, options) {
+        const extent = options.buildExtent ? new Extent(crsOut, Infinity, -Infinity, Infinity, -Infinity) : 0;
+        const coords = readCoordinates(crsIn, crsOut, coordinates, extent);
         const vertices = [];
         for (const c of coords) {
             if (filteringExtent && filteringExtent.isPointInside(c)) {
-                vertices.push(c.xyz());
+                vertices.push(c.rawValues());
             }
         }
-        return { type: 'point', vertices };
+        return { type: 'point', vertices, extent };
     },
 
-    polygon(crsIn, crsOut, coordinates, filteringExtent) {
-        const contour = readCoordinates(crsIn, crsOut, coordinates[0]);
-
+    polygon(crsIn, crsOut, coordinates, filteringExtent, options) {
+        const extent = options.buildExtent ? new Extent(crsOut, Infinity, -Infinity, Infinity, -Infinity) : 0;
+        const contour = readCoordinates(crsIn, crsOut, coordinates[0], extent);
         if (filteringExtent && !filteringExtent.isPointInside(contour[0])) {
             return;
         }
         const vertices2 = [];
-        const vertices = new Array(3 * contour.length);
-        let offset = 0;
-        for (const vertex of contour) {
-            const v = vertex.xyz();
-            v.toArray(vertices, offset);
-            vertices2.push(v);
-            offset += 3;
+        let triangles;
+        if (options.toMesh) {
+            const vertices = new Array(3 * contour.length);
+            let offset = 0;
+            for (const vertex of contour) {
+                const v = vertex.rawValues();
+                v.toArray(vertices, offset);
+                vertices2.push(v);
+                offset += 3;
+            }
+            // TODO: handle holes
+            triangles = Earcut(vertices, null, 3);
+        } else {
+            for (const vertex of contour) {
+                vertices2.push(vertex.rawValues());
+            }
         }
 
-        const triangles = Earcut(vertices, null, 3);
-        // TODO: handle holes
-
-        return { type: 'polygon', vertices: vertices2, indices: triangles };
+        return { type: 'polygon', vertices: vertices2, indices: triangles, extent };
```

Comment:
Don't you need `indices` when `options.toMesh` is false?

Reply:
done